librepo/downloader.c

```diff
@@ -1648,19 +1673,43 @@ check_transfer_statuses(LrDownload *dd, GError **err)
 
             if (!fatal_error &&
                 !complete_url_in_path &&
-                !target->target->baseurl &&
-                (dd->max_mirrors_to_try <= 0 ||
-                 num_of_tried_mirrors < dd->max_mirrors_to_try))
-            {
-                // Try another mirror
-                g_debug("%s: Ignore error - Try another mirror", __func__);
-                target->state = LR_DS_WAITING;
-                g_error_free(transfer_err);  // Ignore the error
-
-                // Truncate file - remove downloaded garbage (error html page etc.)
-                if (!truncate_transfer_file(target, err))
-                    return FALSE;
-            } else {
+                !target->target->baseurl)
+            {   
+              
+                // Another transfers are running or there are successful transfers
+                // and fewer failed transfers than tried parallel connections. It may be mirror is OK
+                // but accepts fewer parallel connections.
+                if (target->mirror->running_transfers > 0 ||
```

Comment:
How about `return has_running_transfers(target) || (has_successful_transfers(target) && !has_less_failed_transfers_than_tried_parallel_connections(target));` in `decide_to_lower_max_tried_parallel_conections()` function?